snapd-glib/requests/snapd-json.c

```diff
@@ -1070,6 +1085,44 @@ SnapdSnap *_snapd_json_parse_snap(JsonNode *node, GError **error) {
   }
   g_ptr_array_add(track_array, NULL);
 
+  JsonObject *links_object = _snapd_json_get_object(object, "links");
+  g_autoptr(GPtrArray) links_array =
+      g_ptr_array_new_with_free_func(g_object_unref);
+  g_autoptr(GPtrArray) link_types = _snapd_json_get_keys(links_object);
+  if (link_types != NULL) {
+    for (guint i = 0; i < link_types->len; i++) {
+      const gchar *key = (const gchar *)link_types->pdata[i];
+      JsonNode *node = json_object_get_member(links_object, key);
+      if (json_node_get_node_type(node) != JSON_NODE_ARRAY) {
+        g_set_error(error, SNAPD_ERROR, SNAPD_ERROR_READ_FAILED,
+                    "Unexpected url type for key '%s'", key);
+        return NULL;
+      }
+
+      g_autoptr(GPtrArray) urls_array = g_ptr_array_new();
+
+      JsonArray *values_array = _snapd_json_get_array(links_object, key);
```

Comment:
`_snapd_json_get_array()` returns a new array, or increments the reference count of the array itself.
```suggestion
      g_autoptr(JsonArray) values_array = _snapd_json_get_array(links_object, key);
```